src/sagemaker/serve/builder/jumpstart_builder.py

```diff
@@ -234,6 +254,213 @@ def _build_for_tgi_jumpstart(self):
 
         self.pysdk_model.env.update(env)
 
+    def _tune_for_js(
+        self, multiple_model_copies_enabled: bool = False, max_tuning_duration: int = 1800
+    ):
+        """Tune for Jumpstart Models.
+
+        Args:
+            multiple_model_copies_enabled (bool): Whether multiple model copies serving is enable by
+                this ``DLC``. Defaults to ``False``
+            max_tuning_duration (int): The maximum timeout to deploy this ``Model`` locally.
+                Default: ``1800``
+        returns:
+            Tuned Model.
+        """
+        if self.mode != Mode.LOCAL_CONTAINER:
+            logger.warning(
+                "Tuning is only a %s capability. Returning original model.", Mode.LOCAL_CONTAINER
+            )
+            return self.pysdk_model
+
+        num_shard_env_var_name = "SM_NUM_GPUS"
+        if "OPTION_TENSOR_PARALLEL_DEGREE" in self.pysdk_model.env.keys():
+            num_shard_env_var_name = "OPTION_TENSOR_PARALLEL_DEGREE"
+
+        initial_env_vars = copy.deepcopy(self.pysdk_model.env)
+        admissible_tensor_parallel_degrees = _get_admissible_tensor_parallel_degrees(
+            self.js_model_config
+        )
+        available_gpus = _get_available_gpus() if multiple_model_copies_enabled else None
+
+        benchmark_results = {}
+        best_tuned_combination = None
+        timeout = datetime.now() + timedelta(seconds=max_tuning_duration)
+        for tensor_parallel_degree in admissible_tensor_parallel_degrees:
+            if datetime.now() > timeout:
+                logger.info("Max tuning duration reached. Tuning stopped.")
+                break
+
+            sagemaker_model_server_workers = 1
+            if multiple_model_copies_enabled:
```

Comment:
what does this do exactly? I thought TGI doesn't support multiple model copies unless used with goldfinch

Reply:
This is anable for DJL. TGI is disabled.

Reply:
DJL does this automatically, do u need this logic?

Reply:
Let's say we have tp degrees [4, 2, 1]

So, with DJL we want
`(4, 1), (2, 2), (1, 4)` as (`tp_degree`, `model copies`) to deploy locally during tuning. Do you mean, DJL will auto deploy model 2 copies during tuning when tp_degree is 2?

If, so, we do not need the logic in the code.